--- wine-5.2.orig/dlls/ntdll/virtual.c	2020-02-17 00:21:52.000000000 +0000
+++ wine-5.2/dlls/ntdll/virtual.c	2020-02-20 06:11:47.149054049 +0000
@@ -2013,15 +2013,23 @@
  */
 void virtual_get_system_info( SYSTEM_BASIC_INFORMATION *info )
 {
+#ifdef __sun
+    LONG64 phys_pages;
+#else
 #ifdef HAVE_SYSINFO
     struct sysinfo sinfo;
 #endif
+#endif /* __sun */
 
     info->unknown                 = 0;
     info->KeMaximumIncrement      = 0;  /* FIXME */
     info->PageSize                = page_size;
     info->MmLowestPhysicalPage    = 1;
     info->MmHighestPhysicalPage   = 0x7fffffff / page_size;
+#ifdef __sun
+    phys_pages = sysconf( _SC_PHYS_PAGES );
+    info->MmHighestPhysicalPage = max(1, phys_pages);
+#else
 #ifdef HAVE_SYSINFO
     if (!sysinfo(&sinfo))
     {
@@ -2029,6 +2037,7 @@
         info->MmHighestPhysicalPage = max(1, total / page_size);
     }
 #endif
+#endif /* __sun */
     info->MmNumberOfPhysicalPages = info->MmHighestPhysicalPage - info->MmLowestPhysicalPage;
     info->AllocationGranularity   = get_mask(0) + 1;
     info->LowestUserAddress       = (void *)0x10000;
